lib/puppet/resource_api.rb

```diff
@@ -517,13 +517,30 @@ def self.try_mungify(type, value, error_msg_prefix)
       # fall through to default handling
     end
 
-    # a match!
-    return [value, nil] if type.instance?(value)
+    is_valid, error_msg = try_validate(type, value, error_msg_prefix)
```

Comment:
Possibly, have try_validate return value, error_msg? If it fails, return nil, error_msg.

Reply:
Doing it like this would obscure the fact that `try_validate` doesn't change the value. But I'll change it to only return `error_msg` or `nil`, which is enough information, and lines up better with `try_mungify`.